core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -747,7 +748,11 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
 
         Query query = multiMatchQuery.parse(type, newFieldsBoosts, value, minimumShouldMatch);
         if (query == null) {
-            return null;
+            if (newFieldsBoosts.isEmpty()) {
```

Comment:
I found two places where I think there is the potential of returning null and moved the checks (and conversion to MatchNoDocs) there. However, its really difficult to check that this cannot happen anywhere, the actual query creating gets delegated around a lot here, so I added an assert to MultiMatchQuery.parse() to at least catch this when we are running tests.